ModuleNotFoundError: No module named 'pkg_resources'

The Jupytext tests that involve `jupyter-fs` started failing on [Feb 12th](https://github.com/mwouts/jupytext/actions/runs/21944330875) when using recent versions of Python (Python 3.12 and above).

As reported at https://github.com/mwouts/jupytext/issues/1509 there is a mention in the logs that `pkg_resources` can't be imported, which in turn causes an empty managers list (see below)

Would you know what might cause the issue? One of the problems that I am facing is that the test passes locally...

Thank you!
```
_____________ test_config_jupytext_jupyter_fs_meta_manager[async] ______________
[gw1] linux -- Python 3.13.13 /home/runner/work/jupytext/jupytext/.pixi/envs/default/bin/python3.13

tmpdir = local('/tmp/pytest-of-runner/pytest-0/popen-gw1/test_config_jupytext_jupyter_f0')
cm_from_fs_meta_manager = <jupytext.async_contentsmanager.build_async_jupytext_contents_manager_class.<locals>.AsyncJupytextContentsManager object at 0x7f8aaa16e270>

    @pytest.mark.asyncio
    async def test_config_jupytext_jupyter_fs_meta_manager(tmpdir, cm_from_fs_meta_manager):
        """Test the configuration of Jupytext with a fs manager"""
        tmpdir.join("jupytext.toml").write('formats = "ipynb,py"')
        cm = cm_from_fs_meta_manager
        # the hash that corresponds to the osfs
>       osfs = [h for h in cm._managers if h != ""][0]
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       IndexError: list index out of range

tests/external/jupyter_fs/test_jupyter_fs.py:83: IndexError
------------------------------ Captured log setup ------------------------------
WARNING  jupyter-fs:metamanager.py:71 Resource {'url': 'osfs:///tmp/pytest-of-runner/pytest-0/popen-gw1/test_config_jupytext_jupyter_f0'} missing 'auth' key, defaulting to 'ask'
WARNING  jupyter-fs:metamanager.py:75 Resource {'url': 'osfs:///tmp/pytest-of-runner/pytest-0/popen-gw1/test_config_jupytext_jupyter_f0', 'auth': 'ask', 'type': 'pyfs'} missing 'type' key, defaulting to 'pyfs'
ERROR    jupyter-fs:metamanager.py:139 Missing dependencies to create manager for resource None
Traceback (most recent call last):
  File "/home/runner/work/jupytext/jupytext/.pixi/envs/default/lib/python3.13/site-packages/jupyterfs/metamanager.py", line 122, in initResource
    managers[_hash] = manager_type.create(
                      ~~~~~~~~~~~~~~~~~~~^
        urlSubbed,
        ^^^^^^^^^^
    ...<5 lines>...
        },
        ^^
    )
    ^
  File "/home/runner/work/jupytext/jupytext/.pixi/envs/default/lib/python3.13/site-packages/jupyterfs/manager/fs.py", line 146, in create
    from fs.errors import FSError
  File "/home/runner/work/jupytext/jupytext/.pixi/envs/default/lib/python3.13/site-packages/fs/__init__.py", line 4, in <module>
    __import__("pkg_resources").declare_namespace(__name__)  # type: ignore
    ~~~~~~~~~~^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'pkg_resources'
```

--- a/README.md
+++ b/README.md
@@ -63,23 +63,23 @@ Add specifications for additional contents managers in your user settings (in th
     {
       "name": "root at test dir",
       "url": "osfs:///Users/foo/test",
-      "type": "pyfs"
+      "type": "fsspec"
     },
     {
       "name": "s3 test bucket",
       "url": "s3://test",
-      "type": "pyfs"
+      "type": "fsspec"
     },
     {
       "name": "s3 test key",
       "url": "s3://test-2/prefix/",
-      "type": "pyfs",
+      "type": "fsspec",
       "defaultWritable": false
     },
     {
       "name": "samba guest share",
       "url": "smb://guest@127.0.0.1/test?name-port=3669",
-      "type": "pyfs"
+      "type": "fsspec"
     }
   ]
 }
@@ -101,7 +101,7 @@ Any stretch of a `"url"` that is enclosed in double-brackets `{{VAR}}` will be t
     {
       "name": "samba share",
       "url": "smb://{{user}}:{{passwd}}@127.0.0.1/test?name-port=3669",
-      "type": "pyfs"
+      "type": "fsspec"
     }
   ]
 }
@@ -153,7 +153,7 @@ export FSSPEC_S3_SECRET=<YOUR SECRET>
 
 ## Choosing the backend
 
-Your resource can include a `"type"` field, set to either `pyfs` or `fsspec`.
+Your resource can include a `"type"` field, set to either `fsspec` or `pyfs`.
 The default is `fsspec`.
 This field can be configured via JSON Settings or graphically.
 

--- a/js/schema/plugin.json
+++ b/js/schema/plugin.json
@@ -51,7 +51,7 @@
         "type": {
           "description": "The backend type to use for the resource (PyFS or fsspec)",
           "type": "string",
-          "enum": ["pyfs", "fsspec"],
+          "enum": ["fsspec", "pyfs"],
           "default": "fsspec"
         },
         "preferredDir": {

--- a/js/src/filesystem.ts
+++ b/js/src/filesystem.ts
@@ -52,7 +52,7 @@ export interface IFSSettingsResource {
   /**
    * Backend type
    */
-  type: "pyfs" | "fsspec";
+  type: "fsspec" | "pyfs";
 
   /**
    * Fallback for determining if resource is writeable. Used only if the underlying PyFilesystem does not provide this information (eg S3)

--- a/jupyterfs/manager/fsspec.py
+++ b/jupyterfs/manager/fsspec.py
@@ -30,6 +30,10 @@ class FSSpecManager(FileContentsManager):
 
         self._default_writable = default_writable
         if isinstance(fs, str):
+            # normalize osfs url to be compatible with fsspec
+            if fs.startswith("osfs://"):
+                fs = fs.replace("osfs://", "file://", 1)
+
             # fs is an fsspec url
             self._fs, root = fsspec.core.url_to_fs(fs, **kwargs)
             self.root = root

--- a/jupyterfs/metamanager.py
+++ b/jupyterfs/metamanager.py
@@ -71,8 +71,8 @@ class MetaManagerShared:
                 self.log.warning("Resource %r missing 'auth' key, defaulting to 'ask'", resource)
                 resource["auth"] = "ask"
             if "type" not in resource or resource["type"] not in ("pyfs", "fsspec"):
-                resource["type"] = "pyfs"
-                self.log.warning("Resource %r missing 'type' key, defaulting to 'pyfs'", resource)
+                resource["type"] = "fsspec"
+                self.log.warning("Resource %r missing 'type' key, defaulting to 'fsspec'", resource)
 
             if resource.get("kwargs", None):
                 if isinstance(resource["kwargs"], str):

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -79,6 +79,7 @@ develop = [
     "fs>=2.4.11",
     "fs-s3fs>=1.1.1",
     "fs.smbfs>=0.6.3",
+    "pkg-resources-backport",
     # fsspec
     "fsspec>=2023.6.0",
     "s3fs>=2024",

